Code/Framework/AzCore/AzCore/Metrics/IEventLogger.h

```diff
@@ -23,8 +24,47 @@ namespace AZ::IO
     class GenericStream;
 }
 
+namespace AZ::Metrics::Internal
+{
+    //! Wraps the Metrics settings prefix key of "/O3DE/Metrics"
+    //! which is used as the parent object of any "/O3DE/Metrics/<EventLoggerName>" keys
+    //! The "/O3DE/Metrics/<EventLoggerName>" is the anchor object where settings
+    //! for an Event Logger are queried.
+    //! Currently  supports the following settings
+    //!
+    //! * "/O3DE/Metrics/<EventLoggerName>/Active" - If set to false, the event logger will not record new events
+    //!   If not set or true, the event logger will record events
+    struct SettingsKey_t
+    {
+        constexpr AZStd::fixed_string<128> operator()(AZStd::string_view name) const
+        {
+            AZStd::fixed_string<128> settingsKey(MetricsSettingsPrefix);
```

Comment:
an Assert check on the size of name could be helpful here so it doesn't overflow the fixed string

Reply:
I can add that